lib/experimental/block-bindings/sources/index.php

```diff
@@ -0,0 +1,24 @@
+<?php
+/**
+ * Define the mechanism to add new sources available in the block bindings API.
+ *
+ * @package gutenberg
+ */
+
+global $block_bindings_sources;
+$block_bindings_sources = array();
```

Comment:
A lot of the registries in WordPress use the approach of having a singleton class, so it could be something to replicate here.

Some examples:
- Block Pattern Categories: https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/class-wp-block-pattern-categories-registry.php
- Block Styles: https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/class-wp-block-styles-registry.php

If a class is used, it's worth thinking about how it might be extended by the gutenberg codebase when shipped in core if changes need to be made to block bindings.

Reply:
Thanks for sharing! @michalczaplinski @artemiomorales It'd be great if you can explore this possibility.

Reply:
Thanks! Agreed that we should begin improving the code as we look to bring this into Core.

In order to keep things moving and start getting more feedback on the feature, and since this is still an experiment, I think we can get this PR merged first then immediately begin another PR to refactor and start moving the logic over to a Block Bindings class 👍

Reply:
Great feedback, @talldan. That’s the direction I would recommend, too. I  agree with @artemiomorales as well that this refactoring might be easier to perform as a focused effort in a follow up PR. All functionality is behind an experiment. Well, another option is to open a new branch targeting this branch if you decide to spend more time polishing this PR.

Some inspiration for how to architect the class and helper functions can be borrowed from the recent work on ES Modules API in https://github.com/WordPress/wordpress-develop/pull/5818 where @felixarntz did an outstanding job explaining several good practices to take into account. That combined with prior work referenced by Dan should give you necessary insights to refactor the code.